blocks/algorithms/__init__.py

```diff
@@ -534,3 +534,59 @@ def compute_steps(self, previous_steps):
             (param, step * multiplier)
             for param, step in previous_steps.items())
         return steps, []
+
+
+class Adam(StepRule):
+    """Adam optimizer as described in [King2014]_.
+
+    .. [King2014] Diederik Kingma, Jimmy Ba,
+       * Adam: A Method for Stochastic Optimization *,
+       http://arxiv.org/abs/1412.6980
+
+    Parameters
+    ----------
+    learning_rate: float, optional
+        Step size.
+        Default value is set to 0.0002.
+    beta_1: float, optional
+        Exponential decay rate for the first moment estimates.
+        Default value is set to 0.1.
+    beta_2: float, optional
+        Exponential decay rate for the second moment estimates.
+        Default value is set to 0.001.
+    epsilon: float, optional
+        Default value is set to 1e-8.
+    decay_factor: float, optional
+        Default value is set to 1e-8.
+
+    """
+    def __init__(self, learning_rate=0.0002,
+                 beta1=0.1, beta2=0.001, epsilon=1e-8,
+                 decay_factor=1e-8):
+        self.learning_rate = learning_rate
+        self.beta1 = beta1
+        self.beta2 = beta2
+        self.epsilon = epsilon
+        self.decay_factor = decay_factor
+
+    def compute_step(self, param, previous_step):
+        mean = shared_floatx(param.get_value() * 0., 'mean')
+        variance = shared_floatx(param.get_value() * 0., 'variance')
+        time = shared_floatx(0., 'time')
+
+        t1 = time + 1
+        learning_rate = self.learning_rate * \
```

Comment:
The line breaks with "\" are discouraged in the Python style guide except for a few cases, so we in Blocks try not to use them. Can you replace those with parentheses?